src/backend/Csrs.Api/Features/Accounts/NewAccountAndFile.cs

```diff
@@ -100,16 +117,25 @@ public async Task<Response> Handle(Request request, CancellationToken cancellati
                 if (request.File.OtherParty != null)
                 {
                     request.File.OtherParty.PartyId = Guid.Empty.ToString();
-                    otherDynamicsParty = request.File.OtherParty.ToDynamicsModel();
+                    otherDynamicsParty = await request.File.OtherParty.ToDynamicsModelAsync(_dynamicsClient, cancellationToken);
                     _logger.LogInformation("Creating other party");
-                    otherDynamicsParty = await _dynamicsClient.Ssgcsrsparties.CreateAsync(body: otherDynamicsParty, cancellationToken: cancellationToken);
+                    try
+                    {
+                        otherDynamicsParty = await _dynamicsClient.Ssgcsrsparties.CreateAsync(body: otherDynamicsParty, cancellationToken: cancellationToken);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, "Exception in Handle.creating other party");
+                    }
+
                 }
                 else
                 {
                     _logger.LogInformation("No other party provided");
```

Comment:
we may want to lower to debug level... is is valid for there to be no other party or is the other party required by the app logic?

Reply:
it will be polished later